### DISCUSSION TITLE
[1주차] 2024-10-15 Scrum Meeting 일지

### DISCUSSION THREAD

**Discussion by yeon-so** [nhnacademy-be7-nuri-books/nuri-books-demo#3] [회의록]
<details>
<summary>목차</summary>
<div markdown="1">

- [001. 사전 협의 사항](#001---------)
  * [팀 이름](#----)
    + [그 외 나온 방안](#---------)
- [002. 요구사항 분석](#002--------)
  * [ERD](#erd)
  * [기타 사항](#-----)
  * [다음 회의 시 필요 사항](#-------------)

</div>
</details>

---

# 001. 개요

## 팀 이름

- 누리의 서재 (NURI Books)

## 그 외 나온 방안

- 팀 규칙
    1. Master 및 서기 주별로 교체
    2. 매일 9시 10분 미팅 시간
- 공유 스페이스 생성
    - git, ERD, Notion(임시)

---

# 002. 요구사항 분석

## ERD

> [!NOTE] 
> 📌__나중에 생각할 점__
>  - [ ]  재고를 도서(book)에 둘 것인지 따로 테이블을 분리할 것인지 (트랜잭션)’
>  - [ ]  주소를 다대다? 일대다?

### 협의 사항
- DB 규칙
    - 컬럼 규칙
        - 무조건 유일한 값
        - (테이블명_컬럼명 금지)
    - 인조키 ID 규칙
        - 무조건 DB 식별 키는 namd : id
        - 단 외래키로 연결 시 [table명]_id
- 회원
    - 비회원 & 회원 구분
        - 사용자 라는 테이블에 회원 구분 없이 등록
        - 단 회원일 경우 회원 테이블 매핑
    - 비밀번호는 사용자가 관리
        - 이메일, 비밀번호 고유
        - 만일 비회원이 같은 정보로 여러 주문을 했을 시 그 정보로 주문한 내역 출력
- 주문
    - 배송 보내는 정보는 사용자(user) 정보랑 매핑
    - 배송 받는 정보는 주문 테이블에 작성
    - 장바구니
        - REDIS 성능 이슈 해소
        - 그리고 REDIS는 클러스터링, 스냅샷으로 In memory (비휘발성) 해결 가능
- 도서
    - 도서(book) Detail 분리 여부
        - - 분리하지 않는다.
    - ISBN 을 PK로 써도 되는가?
        - No, PK를 계좌번호, 카드번호라고 생각해보자.
## 기타 사항

> [!TIP]
> __협의 중 해결된 점__
> 1. 테이블의 식별키를 B00001 처럼 커스텀해서 시퀀스로 두는가?
>    - AutoIncrement를 사용하대 AutoIncrement 역시 트랜젝션을 사용하기 떄문에 성능 영향을 생각하면 사용한다.
> 2. 테이블의 식별키를 B00001 처럼 커스텀해서 시퀀스로 두는가?
>    - AutoIncrement를 사용하대 AutoIncrement 역시 트랜젝션을 사용하기 떄문에 성능 영향을 생각하면 사용한다.
>    - B000001 처럼 커스텀 하는 건 개발 할 때 생각하고 만일 이 형식으로 사용 시 AutoIncrement  와 같이 커스텀 시퀀스 사용 가능하다.
> __그외__
> - 값 암호화 시 Salt 확인 또는 어디서 암호화를 진행할 것 인지 개발 단계에서 확인할 예정

---

# 003. 다음 회의 시 필요 사항
- 우선 순위 순
    - [x]  뷰 템플릿 찾아보기
    - [ ]  코드 스타일 (함수 이름, 예외 처리  등)
    - [x]  매 회의 때 코드 리뷰 여부
    - [ ]  GIt 규칙 설정 (빌드 일시(머지), 커밋 규칙, 양식, PR여부 등등…)
    - [ ]  아키텍처 설계
    - [ ]  DTO 공유 여부

**Comment by yongtaek2** [comment]
누리는 내가 아는 서기 중 제일이다.
